src/schnorrsig.rs

```diff
@@ -211,6 +211,12 @@ impl KeyPair {
         }
     }
 
+    /// Serialize the key pair as a secret key byte value
+    #[inline]
+    pub fn serialize_secret(&self) -> [u8; constants::SECRET_KEY_SIZE] {
+        *SecretKey::from_keypair(self).as_ref()
+    }
```

Comment:
I know that adding this function was the original idea of this PR.

I am wondering on whether or not this is still worth to have given the now very convenient functions of converting a `KeyPair` into a `SecretKey`. Working with `From` and `Into` can sometimes be annoying though - esp. when chaining methods together - because the compiler sometimes requires type annotations.

If serializing a `KeyPair` is the ultimate goal, what about the following API:

```rust
impl KeyPair {
	fn secret_key(&self) -> SecretKey;
	fn public_key(&self) -> PublicKey;
}
```

These are easily accessible, will never require type hints but are more general than providing the secret-key bytes directly. As such, they compose better with the existing functionality. Serializing a `KeyPair` is then just a matter of:

```rust
let bytes = keypair.secret_key().as_ref();
```

Reply:
I overall does not like the current API on getting byte representation of different key types, which uses very distinct names and approaches and not cohesive at all (`serialize*` in one cases, `[..]` or `as_ref()` in other)...

I do not have any preference here other then overhauling the whole key API, which is not a task of this PR. As regarding this PR I will be happy to update it to whatever API will be supported by @apoelstra, @elichai, you and other lib mantainers.